uefi-exts/src/boot.rs

```diff
@@ -33,31 +32,35 @@ impl BootServicesExt for BootServices {
         }
 
         // Perform the search.
-        let (buffer_size, status2) = self.locate_handle(search_type, Some(&mut buffer))?.split();
+        let (status2, buffer_size) = self.locate_handle(search_type, Some(&mut buffer))?.split();
 
         // Once the vector has been filled, update its size.
         unsafe {
             buffer.set_len(buffer_size);
         }
 
+        // Emit output, with warnings
         status1
-            .into_with(|| buffer)
+            .into_with_val(|| buffer)
             .map(|completion| completion.with_status(status2))
     }
 
-    fn find_protocol<P: Protocol>(&self) -> Option<&UnsafeCell<P>> {
+    fn find_protocol<P: Protocol>(&self) -> Result<&UnsafeCell<P>> {
         // Retrieve all handles implementing this.
-        self.find_handles::<P>()
-            // Convert to an option.
-            .warning_as_error()
-            .ok()?
-            // Using the `find_handles` function might not return _only_ compatible protocols.
-            // We have to retrieve them all and find one that works.
```

Comment:
Reading through the documentation of LocateHandle in the UEFI spec, I found nothing that supported this statement.

What I did find, however, was an EFI 1.1 entry point called LocateProtocol that does the same thing as this function, but more efficiently. Considering how old EFI 1.1 is (< 2005), we can probably safely switch to it in the future.

Reply:
I remember having some issues when implementing this function, which is why I took the cautious path and added an extra check. I have no problem with this change if the code still works.

Reply:
Do you mean that you had issues with LocateProtocol specificially? If so, it might be linked to the boot service table struct issue that you fixed in https://github.com/rust-osdev/uefi-rs/commit/035d96b859d79c155f92266c2e25d9cf15bff2e9 . If you want, I can try to add it back in the future...